docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+- for user invitation, this would be the administrators
+- for device invitation, this would be the user alone
+- for shamir invitation, this would be the shamir recipients
+
+> [!NOTE]
+> This set may evolve over time as users can be revoked and user profiles can change.
+
+
+## Goals and Non-Goals
+
+To sum it up, the goal is to design a new invite transport that:
+- allows for network failures by removing the simultaneity constraints of Parsec v2 and using non-blocking HTTP requests
+- adds new semantics to manage attempts distinctly and allow for cancelling them with a reason that's communicated to the other peer
+- decouples the concept of greeter from the invitation author so that a user can be greeted by any administrator
+
+
+## Terminology
+
+Both terms `invite` and `invitation` are used in the code base. There are basically three usage of those words:
+
+1. `invite` as verb, as in "an administrator invites a user to join an organization"
+  * example: the text for `NewUserInvitationError::NotAllowed` is `Not allowed to invite a user`
+
+2. `invite` as a noun, referring to the invite system in general
+  * example: the commands related to the invite system are named `invite_*`
+
+3. `invitation`, referring to a specific invitation that has to managed
+  * example: most invite commands have return status called `invitation_not_found` or `invitation_deleted`
+
+In order to avoid too many changes in the data model, we'll stick to this terminology and apply it in the implementation of this RFC.
+
+## Semantics
+
+This section offers a new semantics proposal, without going into the technical aspects of the exposed routes and their implementations.
+
+Each concept is described with their identifier, immutable properties, internal state, corresponding getters and exposed methods.
+
+
+### Invitation
+
+An invitation is used to verify the identity of a user (called the claimer) on a specific device. The users allowed to verify the identity are called greeters. The invitation can be of 3 types:
+- User invitation
+- Device invitation
+- Shared recovery (shamir) invitation
+
+An invitation is identified by:
+- organization id
+- invitation token
+
+Immutable properties:
+- type (user, device, shamir)
+- created-on timestamp
+- created-by author
+- claimer id:
+  * email address (for user invitation)
+  * or user id (for device or shamir invitation)
+
+Mutable state:
+- state: `PENDING` | `CANCELLED` | `COMPLETED`
+
+Getters:
+- get greeters
+  * for user invitation: all administrators
+  * for device invitation: same as claimer id
+  * for shamir invitation: all recipients
+- get channel for greeter
+
+Methods:
+- create invitation
+- cancel invitation
+- complete invitation
+
+
+### Channel
```

Comment:
Channel is a very broad word (though some would say it's still better than "conduit" :trollface: )

I think it's okay as long as we are always referring to it as "invitation channel" (e.g. having a `invitation_channel` table in the database)

On top of that, the attempt part look more like a channel to me (as data flow through a channel, which is what is done in the attempts)... especially given the very first description of the attempt describes it as "trying to establish a secure communication channel" !

So given the "Channel" thing is about having meeting point between the claimer and a given greeter, why not calling it `appointment` ? (by analogy to e.g. "My wife have an appointment to the l'Haÿ-les-Roses préfecture for her visa, I hope it won't take too long...")

Reply:
45c5cbdd13c257e7879a45fb02b69cb8614a5995